From 5d44e0607e835bd4f9c9ac1bdd7b205363ef8931 Mon Sep 17 00:00:00 2001
From: Anand Balagopalakrishnan <anandb@ti.com>
Date: Sat, 23 Jan 2016 22:48:07 +0530
Subject: [PATCH 1/1] udev-seat: restrict udev enumeration to card0

In case of separate GPU and Display devices as found in embedded systems, we
could have modeset node and render node controlled by different drivers.
There is a distinct possibility that udev enumeration returns the DRM device
corresponding to render node as the primary DRM device.

Obviously, modeset operations cannot be done on the GPU DRM device.

Restrict the udev enumeration to card0 and ensure that DRM device corresponding
to display is returned as the primary DRM device.

Upstream-Status: Pending

Signed-off-by: Anand Balagopalakrishnan <anandb@ti.com>
---
 src/compositor-drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index a44e594..65e6238 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -2812,7 +2812,7 @@ find_primary_gpu(struct drm_backend *b, const char *seat)
 
 	e = udev_enumerate_new(b->udev);
 	udev_enumerate_add_match_subsystem(e, "drm");
-	udev_enumerate_add_match_sysname(e, "card[0-9]*");
+	udev_enumerate_add_match_sysname(e, "card0");
 
 	udev_enumerate_scan_devices(e);
 	drm_device = NULL;
-- 
1.9.1

